### PR TITLE
Fixes #10013: ncf script doesn't call properly list-compatible-inputs properly

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/10013

### CODE REVIEW

**Comment by ncharles** [comment]
Commit modified

File: scripts/ncf [hunk 1]
```diff
@@ -105,6 +199,21 @@ EOF
 
 # MAIN
 # Simple but enough at the moment
+# Check if a server_version is defined
+if [ "$1" = "-V" ]
```
Comment (ncharles) [inline scripts/ncf]: i'm really not convinced by -V to define the version, but I didn't have any better idea

File: scripts/ncf [hunk 2]
```diff
@@ -49,10 +52,36 @@ bundle agent main {
...
 }
 
-# Run the buncle $BUNDLE from cfengine source $CODE
-run() {
-  FILE=$(mktemp)
-  cat > "${FILE}" <<'EOF'
+define_ncf_init() {
+  # If we have a Rudder 3.1, we need to use standard ncf init
+  is_version_valid "$RUDDER_VERSION" "3.1"
+  if [ $? -eq 0 ]
```
Comment (peckpeck) [inline scripts/ncf]: You can just write 
if is_version_valid "$RUDDER_VERSION" "3.1"

Comment (peckpeck) [inline scripts/ncf]:
```
if is_version_valid "$RUDDER_VERSION" "3.1"
then
...
```

File: scripts/ncf [hunk 3]
```diff
@@ -49,10 +52,36 @@ bundle agent main {
...
+  then
+    NCF_VERSION="0.x"
+  fi
+  # If we have a Rudder 3.2 or 4.0, we need to use ncf of version 1
+  is_version_valid "$RUDDER_VERSION" "[3.2 4.0]"
```
Comment (peckpeck) [inline scripts/ncf]: We usually use the form "${VARIABLE}" for variable expansion

File: scripts/ncf (hunk 1 above)
Comment (peckpeck) [inline scripts/ncf]: Not that bad

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**